src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala

```diff
@@ -300,10 +311,12 @@ class MarathonSchedulerService @Inject() (
   }
 
   def abdicateLeadership(): Unit = {
-    log.info("Abdicating")
+    if (leader.get()) {
```

Comment:
In the current (also horrendous version of the code), I applied `synchronized` to many methods. Maybe, you should port that.